compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -202,7 +202,7 @@ public class BallerinaParserErrorHandler extends AbstractParserErrorHandler {
             { ParserRuleContext.VARIABLE_NAME, ParserRuleContext.TYPE_DESC_BEFORE_IDENTIFIER_IN_GROUPING_KEY };
 
     private static final ParserRuleContext[] GROUPING_KEY_LIST_ELEMENT_END =
-            { ParserRuleContext.GROUP_BY_CLAUSE_END, ParserRuleContext.COMMA };
+            { ParserRuleContext.COMMA, ParserRuleContext.GROUP_BY_CLAUSE_END };
```

Comment:
this seems to have no effect on the fix? Exit early path is preferred.

Reply:
This is to improve recovery for the semicolon in the below case

```
function testGroupBy() {
    var res = from var {price1, price2, name} in orders
        group by price1 price2
        select name;
}
```

Previously it was inserting a `where` keyword after `price1`. Hence I feel it is good to prioritize `COMMA` in this case

Reply:
Shall we add a test case then? Otherwise, we have no tests testing this.

Reply:
It's an existing test case. See [query_expr_assert_103.json](https://github.com/ballerina-platform/ballerina-lang/pull/40297/files#diff-7cfc1eb764e1e6411a8bd49726b0fb2b509b5c4c187c16773f3265f6bf58f6b4)